backend/curriculum_tracking/models.py

```diff
@@ -1477,20 +1477,29 @@ class BurndownSnapshot(models.Model):
     def create_snapshot(Cls, user):
         match = Cls.objects.filter(
             user=user,
-            timestamp__gte=timezone.now()
+            timestamp__gte=timezone.now()  
             - timedelta(hours=Cls.MIN_HOURS_BETWEEN_SNAPSHOTS),
         ).first()
-
-        if match:
-            # there was a snapshot taken recently, no need to store another one
-            return match
-
+    
         cards = AgileCard.objects.filter(assignees=user)
         project_cards = cards.filter(content_item__content_type=ContentItem.PROJECT)
 
         complete_cards = cards.filter(status=AgileCard.COMPLETE)
         complete_project_cards = project_cards.filter(status=AgileCard.COMPLETE)
 
+        if match:
+            # there was a snapshot taken recently, it will be updated
+            return Cls.objects.filter(
+                user=user,
+                timestamp__gte=timezone.now()  
+                - timedelta(hours=Cls.MIN_HOURS_BETWEEN_SNAPSHOTS),
+            ).update(
+                cards_total_count=cards.count(),
+                project_cards_total_count=project_cards.count(),
```

Comment:
these update arguments also aren't very DRY. 

How about:

```
snapshot_values = {
       "cards_total_count" : ...
        .....
}
```

Now you can use that in multiple places:

eg:

```
some_query.update(**snapshot_values)

Cls.objects.create(
     user=user,
    **snapshot_values
)
```